python/ray/serve/_private/deployment_state.py

```diff
@@ -1461,29 +1457,31 @@ def _set_target_state(
 
         # Determine if the updated target state simply scales the current state.
         if new_target_state.is_scaled_copy_of(self._target_state):
-
-            curr_num_replicas = self._target_state.target_num_replicas
-            new_num_replicas = new_target_state.target_num_replicas
-            num_replicas_changed = curr_num_replicas != new_num_replicas
-
-            if allow_scaling_statuses and num_replicas_changed:
-                scaling_direction = (
-                    DeploymentStatus.UPSCALING
-                    if new_num_replicas > curr_num_replicas
-                    else DeploymentStatus.DOWNSCALING
+            curr_num = self._target_state.target_num_replicas
+            new_num = new_target_state.target_num_replicas
+
+            if new_num > curr_num:
+                self._curr_status_info = self._curr_status_info.handle_transition(
+                    DeploymentStatusInternalTrigger.AUTOSCALE_UP
+                    if autoscale
+                    else DeploymentStatusInternalTrigger.MANUALLY_INCREASE_NUM_REPLICAS,
+                    message=f"Upscaling from {curr_num} to {new_num} replicas.",
+                    should_autoscale=self.should_autoscale()
+                    and self._is_within_autoscaling_bounds(),
```

Comment:
Would be nice to just define those variables before calling `handle_transition()` to be more readable. There are too much logical complexities stuffed in this statements and making it hard to read 😅

Reply:
This is now automatically solved because of https://github.com/ray-project/ray/pull/41429#discussion_r1434590715!